pkg/framework/aries/default.go

```diff
@@ -46,8 +46,9 @@ func defFrameworkOpts(frameworkOpts *Aries) error {
 		frameworkOpts.storeProvider = storeProv
 	}
 
+	// order is important as DIDExchange service depends on Route service and Introduce depends on DIDExchange
 	frameworkOpts.protocolSvcCreators = append(frameworkOpts.protocolSvcCreators,
-		newExchangeSvc(), newIntroduceSvc(), newRouteSvc())
+		newRouteSvc(), newExchangeSvc(), newIntroduceSvc())
```

Comment:
any reason for the reordering here?

Reply:
Order is important as DIDExchange service depends on Route service and Introduce depends on DIDExchange. Already have a [comment](https://github.com/hyperledger/aries-framework-go/pull/1103/files#diff-da8e5280b0328cb616f0b42390a35b00R49) for this.

Reply:
Ouch.

Reply:
This needs to be fixed. I understand there are "foundational" protocols like did-exchange and maybe routing, but protocols *must* be composable in general. These protocol services shouldn't be coupled like that.

Reply:
We already have an issue for fixing the framework and context relationship https://github.com/hyperledger/aries-framework-go/issues/212. we can handle this issue as part of the larger cleanup.